app/models/collection.rb

```diff
@@ -41,6 +41,7 @@ class Collection < ApplicationRecord
   has_many :supplemental_data_sets, through: :group_viewable_entities, source: :entity, source_type: 'HmisSupplemental::DataSet'
 
   belongs_to :user, optional: true
+  belongs_to :source, optional: true, polymorphic: true
```

Comment:
Out of context I think I would be confused as to what `source` means here as the term is very generic; maybe add some docs (also on user_group)

```suggestion
    # Collections may be dedicated to single "source" entity. These eponymous collections are created through side-effects (see the EntityAccess concern). However they may be managed by admins like other collections
    belongs_to :source, optional: true, polymorphic: true
```